tests/lib/nested.sh

```diff
@@ -242,32 +253,41 @@ refresh_to_new_core(){
     fi
 }
 
-get_snakeoil_key(){
+get_snakeoil_key() {
     local KEYNAME="PkKek-1-snakeoil"
     wget https://raw.githubusercontent.com/snapcore/pc-amd64-gadget/20/snakeoil/$KEYNAME.key
     wget https://raw.githubusercontent.com/snapcore/pc-amd64-gadget/20/snakeoil/$KEYNAME.pem
     echo "$KEYNAME"
 }
 
-secboot_sign_gadget(){
+secboot_sign_gadget() {
     local GADGET_DIR="$1"
     local KEY="$2"
     local CERT="$3"
     sbattach --remove "$GADGET_DIR"/shim.efi.signed
     sbsign --key "$KEY" --cert "$CERT" --output pc-gadget/shim.efi.signed pc-gadget/shim.efi.signed
 }
 
-cleanup_nested_env(){
-    rm -rf "$WORK_DIR"
+prepare_nested_env() {
+    mkdir -p "$NESTED_IMAGES_DIR"
+    mkdir -p "$NESTED_RUNTIME_DIR"
+    mkdir -p "$NESTED_ASSETS_DIR"
+    mkdir -p "$NESTED_LOGS_DIR"
 }
 
-get_image_dir(){
-    echo "$WORK_DIR/image"
+cleanup_nested_env() {
+    rm -rf "$NESTED_RUNTIME_DIR"
+    rm -rf "$NESTED_ASSETS_DIR"
+    rm -rf "$NESTED_LOGS_DIR"
+    rm -rf "$NESTED_IMAGES_DIR"/*.img
+    rm -rf "$NESTED_IMAGES_DIR/$(get_current_image_name).xz"
+    prepare_nested_env
```

Comment:
why does `cleanup` call `prepare`? Seems counter-intuitive to me to call a cleanup function which then cleans up and then re-prepares? If that's how it's meant to be used, perhaps `reset_nested_env` is a better name here for this function

Reply:
I agree, it should not do this.

Reply:
I removed that, thanks for the comment